### PR TITLE
Flow Windows Certificate Store support added.

### LINKED ISSUE #4887: Flow: support TLS from the Windows certificate store like in static mode
### Request

Static mode supports using the Windows certificate store for running the TLS server, where the TLS keypair is stored securely in Windows and encryption is deferred to the Windows certificate store using Windows' API.

As part of feature parity with static mode, Flow mode also needs this functionality.

### Use case

To run a TLS server on Windows without storing the TLS keypair as plain files.

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Add support for windows certificate filter in flow, also remove restrictions in static mode.
 
#### Which issue(s) this PR fixes

Closes #4887
Closes #5176

#### Notes to the Reviewer

I did not add config converters because this is TLS and that`s primarily driven via command line flags.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [X] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/reference/config-blocks/http.md [hunk 1]
```diff
@@ -150,3 +156,51 @@ The following versions are recognized:
 * `TLS12` for TLS 1.2
 * `TLS11` for TLS 1.1
 * `TLS10` for TLS 1.0
+
+
+### windows certificate filter block
+
+The `windows_certificate_filter` block is used to configure retrieving certificates from the built-in Windows
+certificate store. When using the `windows_certificate_filter` block
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/config-blocks/http.md:164]:
```suggestion
certificate store. When you use the `windows_certificate_filter` block
```

File: docs/sources/flow/reference/config-blocks/http.md [hunk 2]
```diff
@@ -150,3 +156,51 @@ The following versions are recognized:
...
+
+* `cert_pem`
+* `cert_file`
+* `key_pem`
+* `key_file`
+* `client_ca`
+* `client_ca_file`
+
+{{% admonition type="warning" %}}
+This feature is not available on any platforms other than Windows.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/config-blocks/http.md:175]:
```suggestion
This feature is only available on Windows.
```

File: docs/sources/flow/reference/config-blocks/http.md [hunk 3]
```diff
@@ -150,3 +156,51 @@ The following versions are recognized:
...
+
+### server block
+
+The `server` block is used to find the certificate to check the signer. If multiple certificates are found the 
+`windows_certificate_filter` will choose the certificate with the expiration farthest in the future.
+
+Name | Type           | Description                                                                               | Default | Required
+---- |----------------|-------------------------------------------------------------------------------------------|---------| --------
+`store` | `string`       | Name of the system store to look for the server Certificate ex LocalMachine, CurrentUser. | `""`    | yes
+`system_store` | `string`       | Name of the store to look for the server Certificate ex My, CA.                           | `""`    | yes
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/config-blocks/http.md:190]:
```suggestion
`store` | `string`       | Name of the system store to look for the server Certificate, for example, LocalMachine, CurrentUser. | `""`    | yes
`system_store` | `string`       | Name of the store to look for the server Certificate, for example, My, CA.                           | `""`    | yes
```

Do we want/need to include examples here? Generally these are noted below the table.

File: docs/sources/flow/reference/config-blocks/http.md [hunk 4]
```diff
@@ -150,3 +156,51 @@ The following versions are recognized:
...
+`issuer_common_names` | `list(string)` | Issuer common names to check against.                                                     |         | no
+`template_id` | `string`       | Server Template ID to match in ASN1 format ex "1.2.3".                                    | `""`    | no
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/config-blocks/http.md:192]:
```suggestion
`template_id` | `string`       | Server Template ID to match in ASN1 format, for example, "1.2.3".                                    | `""`    | no
```

File: docs/sources/flow/reference/config-blocks/http.md [hunk 5]
```diff
@@ -150,3 +156,51 @@ The following versions are recognized:
...
+
+### client block
+
+The `client` block is used to check the certificate presented to the server.
+
+Name | Type           | Description                                            | Default | Required
+---- |----------------|--------------------------------------------------------|-----| --------
+`issuer_common_names` | `list(string)` | Issuer common names to check against.                  |     | no
+`subject_regex` | `string`       | Regular expression to match Subject name.              | `""` | no
+`template_id` | `string`       | Client Template ID to match in ASN1 format ex "1.2.3".                 |   `""`   | no
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/config-blocks/http.md:205]:
```suggestion
`template_id` | `string`       | Client Template ID to match in ASN1 format, for example, "1.2.3".                 |   `""`   | no
```

**Review by clayton-cornell — COMMENTED** [review]
Some doc suggestions

**Review by clayton-cornell — COMMENTED** [review]
Docs look OK

File: service/http/tls.go [hunk 6]
```diff
@@ -25,14 +27,75 @@ type TLSArguments struct {
...
+func (args *TLSArguments) validateWindowsCertificateFilterTLS() error {
+	switch {
+	case len(args.Cert) > 0:
+	case len(args.Key) > 0:
+	case len(args.CertFile) > 0:
+	case len(args.ClientCA) > 0:
+	case len(args.ClientCAFile) > 0:
+	case len(args.KeyFile) > 0:
+		return fmt.Errorf("cannot specify any key, certificate or CA when using windows certificate filter")
+	}
```
Comment (wildum) [inline service/http/tls.go:83]: there is no fallthrough by default in Go, so unless Im mistaken the error is returned only when len(args.KeyFile) is > 0 but in the other cases it's not returning the error?
Maybe you meant this instead:
```
switch {
case len(args.Cert) > 0,
     len(args.Key) > 0,
     len(args.CertFile) > 0,
     len(args.ClientCA) > 0,
     len(args.ClientCAFile) > 0,
     len(args.KeyFile) > 0:
	return fmt.Errorf("cannot specify any key, certificate or CA when using windows certificate filter")
}
```

**Review by erikbaranowski — APPROVED** [review]
LGTM pending the other small feedback

**Pull request merged**